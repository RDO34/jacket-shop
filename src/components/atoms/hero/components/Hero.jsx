import React from 'react';
import styled from '@emotion/styled';

export const Hero = props => {
  const { src, height, children } = props;
  if (src) {
    const HeroImage = styled.div`
      width: 100%;
      height: ${height || '512'}px;
      background-image: url(${src});
    `;
    return (
      <HeroImage data-testid="hero-atom" {...props}>
        {children}
      </HeroImage>
    );
  }
  return null;
};
