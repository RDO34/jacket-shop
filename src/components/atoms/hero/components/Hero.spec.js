import React from 'react';
import { render } from 'react-testing-library';

import { Hero } from './Hero';

describe('Hero', () => {
  it('renders the Hero component if passed an image src', () => {
    const { queryByTestId } = render(<Hero src="/some-source/image.jpeg" />);
    expect(queryByTestId('hero-atom')).toBeTruthy();
  });

  it('does not render the Hero component if it is not passed an image src', () => {
    const { queryByTestId } = render(<Hero />);
    expect(queryByTestId('hero-atom')).toBeFalsy();
  });

  it('renders children', () => {
    const ChildComponent = () => <div data-testid="child-component" />;
    const { queryByTestId } = render(
      <Hero src="/some-source/image.jpeg">
        <ChildComponent/>
      </Hero>);
    expect(queryByTestId('child-component')).toBeTruthy();
  });
});
