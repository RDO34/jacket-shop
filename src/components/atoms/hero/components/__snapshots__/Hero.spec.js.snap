// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero snapshot tests renders with custom props 1`] = `
<div
  class="css-cfvpbv"
  data-testid="some-testid"
  height="240px"
  src="some-src"
>
  <div
    data-testid="child-component"
  />
</div>
`;

exports[`Hero snapshot tests renders with default props 1`] = `
<div
  class="css-1llnn4a"
  data-testid="hero-atom"
  src="some-src"
/>
`;
